overlord/devicestate/devicemgr.go

```diff
@@ -885,10 +885,29 @@ func (m *DeviceManager) RequestSystemAction(systemLabel string, action SystemAct
 		return ErrUnsupportedAction
 	}
 
-	// TODO:UC20 assume we are in the run mode
-	if sysAction.Mode == "run" {
-		// do nothing
-		return nil
+	switch m.systemMode {
+	case "recover", "run":
+		// if going from recover to recover or from run to run, do nothing
+		if m.systemMode == sysAction.Mode {
```

Comment:
shouldn't this check that the systemLabel is the same?

Reply:
yes you're right it should

Reply:
well also note that we only will get a sysAction of recover/run if the requested system is the same as the current system from the `systemFromSeed` function, so it's not possible currently to have non-matching labels for an action of recover or run, but perhaps some-day that will change in which case the check you mention will be necessary

Reply:
ah, I see, I forgot that we cross check that way. Do we have a test for it ?

Reply:
anyways I added this check too